yt_dlp/extractor/nhk.py

```diff
@@ -492,3 +495,74 @@ class NhkRadioNewsPageIE(InfoExtractor):
         return self.url_result('https://www.nhk.or.jp/radio/ondemand/detail.html?p=F261_01', NhkRadiruIE)
+
+
+class NhkRadiruLiveIE(InfoExtractor):
+    _GEO_COUNTRIES = ['JP']
+    _VALID_URL = r'https?://www\.nhk\.or\.jp/radio/player/\?ch=(?P<id>r[12]|fm)'
+    _TESTS = [{
+        # radio 1, no area specified
+        'url': 'https://www.nhk.or.jp/radio/player/?ch=r1',
+        'info_dict': {
+            'id': 'r1-tokyo',
+            'title': 're:^ＮＨＫネットラジオ第1 東京.+$',
+            'ext': 'm4a',
+            'thumbnail': 'https://www.nhk.or.jp/common/img/media/r1-200x200.png',
+            'live_status': 'is_live',
+        },
+    }, {
+        # radio 2, area specified
+        # (the area doesnt actually matter, r2 is national)
+        'url': 'https://www.nhk.or.jp/radio/player/?ch=r2',
+        'params': {'extractor_args': {'nhkradirulive': {'area': ['fukuoka']}}},
+        'info_dict': {
+            'id': 'r2-fukuoka',
+            'title': 're:^ＮＨＫネットラジオ第2 福岡.+$',
+            'ext': 'm4a',
+            'thumbnail': 'https://www.nhk.or.jp/common/img/media/r2-200x200.png',
+            'live_status': 'is_live',
+        },
+    }, {
+        # fm, area specified
+        'url': 'https://www.nhk.or.jp/radio/player/?ch=fm',
+        'params': {'extractor_args': {'nhkradirulive': {'area': ['sapporo']}}},
+        'info_dict': {
+            'id': 'fm-sapporo',
+            'title': 're:^ＮＨＫネットラジオＦＭ 札幌.+$',
+            'ext': 'm4a',
+            'thumbnail': 'https://www.nhk.or.jp/common/img/media/fm-200x200.png',
+            'live_status': 'is_live',
+        }
+    }]
+
+    _NOA_STATION_IDS = {'r1': 'n1', 'r2': 'n2', 'fm': 'n3'}  # lookup table is easier than working it out
+
+    def _real_extract(self, url):
+        station = self._match_id(url)
+        area = traverse_obj(self._configuration_arg('area'), 0) or "tokyo"
+
+        config = self._download_xml('https://www.nhk.or.jp/radio/config/config_web.xml', None, 'Downloading area information')
+        data = config.find(f'.//data//area[.="{area}"]/..')  # a <data> element with an <area> that matches the ^variable
+
+        if data is None:
+            raise ExtractorError('Invalid area - valid areas are: %s'
+                                 % ', '.join([i.text for i in config.findall('.//data//area')]),
+                                 expected=True)
+
+        noa_info = self._download_json(f'https:{config.find(".//url_program_noa").text}'
+                                       .replace("{area}", data.find('areakey').text), station)
+
+        present_info = traverse_obj(noa_info, ('nowonair_list', self._NOA_STATION_IDS.get(station), 'present'))
+        service = present_info.get('service')
+
+        return {
+            'title': join_nonempty(service.get('name'), traverse_obj(present_info, ('area', 'name')), delim=" "),
+            'id': join_nonempty(station, area),
+            'thumbnails': traverse_obj(service, ('images', ..., {
```

Comment:
we can simplify `title` extraction with a single traversal instead, and eliminate the need for the `service` var
```suggestion

        return {
            'title': ' '.join(traverse_obj(present_info, (('service', 'area',), 'name', {str}))),
            'id': join_nonempty(station, area),
            'thumbnails': traverse_obj(present_info, ('service', 'images', ..., {
```
also, is `join_nonempty` needed for `station` and `area`? will it ever be the case that one of them is empty?

Reply:
> will it ever be the case that `station` or `area` is empty?

probably not, it's just more convenient to use the nice util function imo